src/org/openstreetmap/josm/gui/dialogs/relation/actions/AddFromSelectionAction.java

```diff
@@ -30,27 +30,40 @@ protected boolean isPotentialDuplicate(OsmPrimitive primitive) {
     }
 
     protected List<OsmPrimitive> filterConfirmedPrimitives(List<OsmPrimitive> primitives) throws AddAbortException {
+        return filterConfirmedPrimitives(primitives, false);
+    }
+
+    protected List<OsmPrimitive> filterConfirmedPrimitives(List<OsmPrimitive> primitives, boolean abortOnSkip) throws AddAbortException {
```

Comment:
Can you please add javadocs here?
```suggestion
    /**
     * Filter confirmed primitives
     * @param primitives The primitives to filter on
     * @param abortOnSkip If the user decides to not add a primitive, abort (throw {@code AddAbortException})
     * @return The primitives to add to the relation. Never {@code null}, but may be an empty list.
     * @throws AddAbortException when {@code abortOnSkip} is {@code true} <i>and</i> the user decides to not add a primitive.
     * @since xxx
     */
    protected List<OsmPrimitive> filterConfirmedPrimitives(List<OsmPrimitive> primitives, boolean abortOnSkip) throws AddAbortException {
```

You'll want to check this with `ant pmd checkstyle`, but it should be right.

As a side note, I'd prefer to make this method more generic (e.g. `protected <O extends IPrimitive> List<O> filterConfirmedPrimitives(List<O> primitives, boolean abortOnSkip) throws AddAbortException`), but I don't know how hard it would be to do (I suspect there would be a cascading chain of methods to change).